dipy/testing/decorators.py

```diff
@@ -103,3 +104,26 @@ def _set_random_number_generator_wrapper(pytestconfig, *args, **kwargs):
         return _set_random_number_generator_wrapper
 
     return _set_random_number_generator
+
+def warn_on_positional_args(valid=""):
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            sig = inspect.signature(func)
+            param_names = list(sig.parameters.keys())
+
+            # Determine the keyword-only arguments
+            kwonly_args = [name for name, param in sig.parameters.items() 
+                           if param.kind == param.KEYWORD_ONLY]
+
+            # Check the number of positional arguments
+            num_positional_args = len(args)
+            for kwarg in kwonly_args:
+                if kwarg in param_names[:num_positional_args]:
+                    warning_msg = (f"Warning: Function '{func.__name__}' keyword-only argument '{kwarg}' is being "
+                                   f"called positionally. This will be deprecated from version '{valid}'.")
+                    warnings.warn(warning_msg, stacklevel=2)
+            
+            return func(*args, **kwargs)
```

Comment:
In the decorated function `func`, if there are N positional arguments but N+M (M > 0) are given, this will result in a `TypeError: func takes N positional arguments but N+M were given`, so either we `try` the function and catch the exception reorganizing the arguments and giving a warning about bad usage (could lead to unexpected function result for the user); or we update the tests to catch the `TypeError` when this happens (this is simpler).

Reply:
I agree. Thanks for the feedback